lib/batch/batch_metadata_updates_655.rb

```diff
@@ -0,0 +1,27 @@
+# Transfer metadata values from Original Publisher to Digital Publisher field
+
+# below is too slow
+#items_needing_transfer = GenericFile.all.reject { |gf| 
+#                           gf.original_publisher == []
+#                         }
+
+solr_items_w_original_publisher_value = ActiveFedora::SolrService.query(
+                                     'original_publisher_tesim:*', 
+                                     { rows: 99999 }
+                                   )
+
+solr_items_w_original_publisher_value.each do |solr_item|
+  # could be either Collection or GenericFile, so we convert the string to
+  # class to get the object by id
+  constant_item = solr_item['has_model_ssim'].first.constantize
+  item = constant_item.find(solr_item['id'])
```

Comment:
Is this a DB call or a Solr call? How slow is this script? Have you looked into Solr/DB's bulk update?

Reply:
It's an ActiveFedora call, and does not have a bulk update. This isn't slow, but it's not super fast. Querying the index was instant, and doing a `.find()` for each GenericFile or Collection was faster than getting all the ActiveFedora objects, then rejecting ones with empty `original_publisher` fields. 

The slow comment I made was in reference to `GenericFile.all.reject`

Reply:
> It's an ActiveFedora call

... which is a DB or a Solr call ;) ? I *think* it is a DB call. I *think* ActiveFedora provides an ActiveSupport like ORM over Fedora's entries. But then why would this approach be any faster? Unless `.all` loads *everything* in memory and this chokes your system.

Reply:
Yes, you are right about all of this. In my mind I wasn't sure to call Fedora's entries a DB object, though ActiveFedora is based on ActiveSupport like ORM for Fedora. With that said, the `.all` does overload the memory, which is why this approach is better.